### PR TITLE
Pick and Place sound effects and improvemets

### PR DESCRIPTION
**Pull request by robhillman97**
Added sound effects to pick and place to address #795. 

Also the arrow object size is now saved with other pick and place data and bounding boxes have been added to the arrow and sphere

### CODE REVIEW

File: Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs [hunk 1]
```diff
@@ -91,6 +98,10 @@ void Update()
                 }
                 transform.hasChanged = false;
             }
+            else if(moveMode && pickOb.GetComponentInChildren<Renderer>().material.color != originalArrowColor)
```
Comment (cawke) [inline Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs]: cache this value - `pickOb.GetComponentInChildren<Renderer>()` and use it instead.

File: Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs [hunk 2]
```diff
@@ -178,26 +191,42 @@ public void ManipulationStart()
...
-            {
-                pickOb.transform.localPosition = new Vector3(resetPos.x, resetPos.y, resetPos.z);
+                {
+ 
+                    pickOb.transform.localPosition = new Vector3(placeLocation.localPosition.x, placeLocation.localPosition.y, placeLocation.localPosition.z);
 
+                    pickOb.GetComponentInChildren<Renderer>().material.SetColor("_Color", Color.green);
+                   
+                    if (shouldPlaySound)
+                        playAudio(correctAudio);
```
Comment (cawke) [inline Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs:210]: add brackets

File: Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs [hunk 3]
```diff
@@ -178,26 +191,42 @@ public void ManipulationStart()
...
 
-                pickOb.GetComponentInChildren<Renderer>().material.SetColor("_Color", originalArrowColor);
-            }
+                else if (resetOnMiss)
+                {                  
+                        pickOb.transform.localPosition = resetPos;
 
+                        pickOb.GetComponentInChildren<Renderer>().material.SetColor("_Color", originalArrowColor);
+                        if (shouldPlaySound)
+                            playAudio(incorrectAudio);
```
Comment (cawke) [inline Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs]: add brackets

**Review by cawke — CHANGES_REQUESTED** [review]

File: Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs (hunk 1 above)
Comment (Alvareon) [reply Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs]: yeah, we should avoid is calling GetComponent every frame if we don't have to

Comment (Alvareon) [reply Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/Pick.cs]: yeah, we should avoid is calling GetComponent every frame if we don't have to

**Review by cawke — APPROVED** [review]
Thanks

**Pull request merged**